includes/test-user.php

```diff
@@ -0,0 +1,168 @@
+        add_filter('user_row_actions', [$this, 'ppc_test_user_row_actions'], 10, 2);
+        //add return message notice link
+        add_action('wp_head', [$this, 'ppc_test_user_revert_notice']);
+        add_action('all_admin_notices', [$this, 'ppc_test_user_revert_notice']);
+
+        $this->ppc_test_user_action();
+    }
+
+    /**
+     * Initialize test user component
+     */
+    public static function init()
+    {
+        $instance = new self;
+        return $instance;
+    }
+
+    /**
+     * Adds Test this user to users row action.
+     *
+     * @param $actions
+     * @param $user
+     * 
+     * @return $action
+     */
+    public function ppc_test_user_row_actions($actions, $user)
+    {
+        if (current_user_can('manage_capabilities') && current_user_can('edit_user', $user->ID) && $user->ID !== get_current_user_id()) {
+
+            $link = add_query_arg(
+                [
+                    'ppc_test_user' => base64_encode($user->ID), 
+                    '_wpnonce'      => wp_create_nonce('ppc-test-user')
+                ], 
+                admin_url('users.php')
+            );
+
+            $actions['ppc_test_user'] = sprintf(
+                '<a href="%s">%s</a>',
+                esc_url($link),
+                esc_html__('Test this user', 'capsman-enhanced')
+            );
+        }
+
+        return $actions;
+    }
+
+    /**
+     * Test user process
+     */
+    public function ppc_test_user_action()
+    {
+
+        global $current_user;
+            
+        if (!is_user_logged_in() || !isset($_GET['ppc_test_user']) || !isset($_GET['_wpnonce'])) {
+            return;
+        }
+
+        if (!wp_verify_nonce(sanitize_key($_GET['_wpnonce']), 'ppc-test-user')) {
+            wp_die(esc_html__('Your link has expired, refresh the page and try again.', 'capsman-enhanced'));
+        }
+
+        $request_user_id = isset($_GET['ppc_test_user']) ? (int) base64_decode(sanitize_text_field($_GET['ppc_test_user'])) : 0;
+        $ppc_return_back = isset($_GET['ppc_return_back']) ? (int) sanitize_text_field($_GET['ppc_return_back']) : 0;
+        $request_user    = get_userdata($request_user_id);
+        
+        if (!$request_user || (is_object($request_user) && !isset($request_user->ID))) {
+            wp_die(esc_html__('Unable to retrieve user data.', 'capsman-enhanced'));
+        } else {
+            if ($ppc_return_back > 0) {
+                $user_auth        = wp_unslash(self::ppc_test_user_tester_auth());
+                $original_user_id = wp_validate_auth_cookie($user_auth, 'logged_in');
+                if ($original_user_id) {
+                    wp_set_auth_cookie($original_user_id, false);
+                    // Unset the cookie
+                    // phpcs:ignore WordPressVIPMinimum.Functions.RestrictedFunctions.cookies_setcookie
+                    setcookie('ppc_test_user_tester_'.COOKIEHASH, 0, time()-3600, COOKIEPATH, COOKIE_DOMAIN, is_ssl(), true);
```

Comment:
These numbers for time offset could be more readable if you have a constant in the class for them, like the constant `HOUR_IN_SECONDS`. This makes a lot more clear and easier to maintain in the future.

Reply:
Please, check here: https://codex.wordpress.org/Easier_Expression_of_Time_Constants